src/lit-html.ts

```diff
@@ -12,21 +12,39 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
+/**
+ * TypeScript has a problem with precompiling templates literals
+ * https://github.com/Microsoft/TypeScript/issues/17956
+ */
+const shouldCacheTemplates = ((_t: any) => _t() === _t())(() => ((s: any) => s)``);
+
 // The first argument to JS template tags retain identity across multiple
 // calls to a tag for the same literal, so we can cache work done per literal
 // in a Map.
 const templates = new Map<TemplateStringsArray, Template>();
+const cacheTemplates = new Map<string, TemplateStringsArray>();
```

Comment:
We don't need a separate template cache, just change the key type on the first to `TemplateStringsArray|string`